tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -41,27 +41,35 @@ limitations under the License.
 #include "tensorflow/core/platform/logging.h"
 #include "tensorflow/core/util/env_var.h"
 
+#if TENSORFLOW_USE_ROCM
+#include "rocm/include/hip/hip_runtime.h"
+#endif
+
 namespace tensorflow {
 namespace grappler {
 
 #if TENSORFLOW_USE_ROCM
-const std::array<std::string, 2> FP16SupportedDevices = {"906", "908"}; 
-
-bool HasEnhancedFP16ComputeSupport(std::pair<int, int> gpu_arch){
-    std::string arch = std::to_string(gpu_arch.first); 
-    return std::find(std::begin(FP16SupportedDevices), 
-                     std::end(FP16SupportedDevices), arch)
-           != std::end(FP16SupportedDevices); 
+bool GetFastFP16Support(const DeviceProperties &props)
+{
+    bool supported = false;
+    std::array<std::string, 2> FP16SupportedDevices = {"gfx906", "gfx908"};
+    std::string gcnArchName = props.environment().at("architecture");
+#if TF_ROCM_VERSION >= 4000
+    std::string gpu_arch = absl::StrSplit(gcnArchName, ":")[0];
+#else
+    std::string gpu_arch = gcnArchName;
+#endif
```

Comment:
double check locally, but I think you can make the code within the `#if` unconditional.  `absl::StrSplit(gcnArchName, ":")[0]` should return you the gcnArchName, even when there is no ":" present

Reply:
I believe you are correct.